### PR TITLE
[NFC] Spin Off Dependency Tracing Utilities From FrontendTool

### PR DESCRIPTION
**Pull request by CodaFi**
This leaves FrontendTool devoid of everything except the high-level entrypoints to functionality in the rest of the compiler stack. Hopefully we can make a later pass and drop a bunch of unused includes...

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Review by davidungar — APPROVED** [review]
LGTM! I like the constification. There were one or two places where a const reference could maybe be used.

File: lib/AST/FrontendSourceFileDepGraphFactory.h [hunk 1]
```diff
@@ -23,29 +23,29 @@ namespace fine_grained_dependencies {
 
 class FrontendSourceFileDepGraphFactory
     : public AbstractSourceFileDepGraphFactory {
-  SourceFile *const SF;
+  const SourceFile *SF;
```
[the remark this answers is not in the document]
Comment (CodaFi) [reply lib/AST/FrontendSourceFileDepGraphFactory.h:26]: It would. But if the goal is to prevent reassignment, encapsulation is your best bet.

**Comment by CodaFi** [comment]
@swift-ci smoke test Linux

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test macOS

**Comment by CodaFi** [comment]
⛵

**Pull request merged**